OpenRA.Game/Graphics/TerrainSpriteLayer.cs

```diff
@@ -50,21 +50,21 @@ public TerrainSpriteLayer(World world, WorldRenderer wr, Sheet sheet, BlendMode
 			vertexBuffer = Game.Renderer.Device.CreateVertexBuffer(vertices.Length);
 			emptySprite = new Sprite(sheet, Rectangle.Empty, TextureChannel.Alpha);
 
-			wr.PaletteInvalidated += () =>
-			{
-				paletteIndex = palette.TextureIndex;
+			wr.PaletteInvalidated += ChangePaletteIndexes;
+		}
 
-				// Everything in the layer uses the same palette,
-				// so we can fix the indices in one pass
-				for (var i = 0; i < vertices.Length; i++)
-				{
-					var v = vertices[i];
-					vertices[i] = new Vertex(v.X, v.Y, v.Z, v.U, v.V, paletteIndex, v.C);
-				}
+		void ChangePaletteIndexes()
```

Comment:
We use "Indices" in other parts of the code, but TIL that "Indexes" is valid / preferred in en-us.
It would be nice to keep this consistent with our other code.